lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -1521,7 +1523,7 @@ createSpecializedStructOrClassType(NominalOrBoundGenericNominalType *Type,
     SmallVector<llvm::Metadata *, 16> TemplateParams;
     for (auto Arg : GenericArgs) {
       DebugTypeInfo ParamDebugType;
-      if (Opts.DebugInfoLevel > IRGenDebugInfoLevel::ASTTypes)
+      if (Opts.DebugInfoLevel > IRGenDebugInfoLevel::ASTTypes && !llvm::isa<GenericTypeParamType>(Arg) && !AsForwardDeclarations)
```

Comment:
`!llvm::isa<GenericTypeParamType>(Arg)` should be written as `!Arg->is<GenericTypeParamType>()`, but that sounds like the wrong condition. What are you trying to determine here?

Reply:
Re-examining this the condition was incorrect, I've removed it.